### PR TITLE
[Pods] Fleshing out PodDebugTabView

### PR DESCRIPTION
**Pull request by wavesoft**
---
~~⚠️ Requires #1071 to be merged~~ Merged!

---

This PR implements the DebugTab view. It shows the time of the last change, and it also extracts last termination information according to this spec: https://github.com/mesosphere/marathon/blob/feature/pods/docs/docs/rest-api/public/api/v2/types/podStatus.raml#L217

image

### CODE REVIEW

**Comment by jfurrow** [comment]
I'd suggest moving the `Containers` row to its own section, so that the the table can fill the width of the page, rather than the column it's in currently. Here's my markdown representation of the idea:

#### Last Changes
```
Configuration   Wed...
```

#### Last Terminations
```
Terminated at...
Instance ID     pod...
Message         Pod...
Started At      Wed...
```

#### Containers
```
Containers table goes here
```

File: src/js/components/PodDetail.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      </pre>
+    );
+  }
+
+  render() {
+    const {pod} = this.props;
+
+    return (
+      <div className="flex-container-col">
+        <div className={classNames(
```
Comment (jfurrow) [inline src/js/components/PodDetail.js]: Let's just pass a string with all of these classnames since we're not conditionally applying any of them.

**Comment by wavesoft** [comment]
Note that containers is an array on the last termination (that on it's an array on it's own).

File: src/js/components/PodDetail.js (hunk 1 above)
Comment (wavesoft) [inline src/js/components/PodDetail.js]: This was a solution to break the long lines. Should I completely discard this?

**Comment by wavesoft** [comment]
Regarding https://github.com/dcos/dcos-ui/pull/1082#issuecomment-246751759, I am in a discussion with @leemunroe to find out a nice solution to this. 

For now I moved the table at the end of the list:

image

File: src/js/components/PodContainerTerminationTable.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  render() {
+    return (
+        <Table
+          className={this.props.className}
+          columns={this.getColumns()}
+          colGroup={this.getColGroup()}
+          data={this.props.containers}
+          itemHeight={TableUtil.getRowHeight()}
+          sortBy={{prop: 'ID', order: 'desc'}}
+          />
```
Comment (mlunoe) [inline src/js/components/PodContainerTerminationTable.js]: #nit Let's move this up to be `sortBy={{prop: 'ID', order: 'desc'}} />`

File: src/js/components/PodDebugTabView.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+import Pod from '../structs/Pod';
+import PodContainerTerminationTable from './PodContainerTerminationTable';
+import TimeAgo from './TimeAgo';
+
+class PodDebugTabView extends React.Component {
+  getTerminationHistory() {
+    let history = this.props.pod.getTerminationHistoryList().getItems();
+    if (!history.length) {
+      return (
+          <div>(No data)</div>
```
Comment (mlunoe) [inline src/js/components/PodDebugTabView.js]: #nit indentation

File: src/js/components/PodDetail.js [hunk 4]
```diff
@@ -40,10 +41,8 @@ class PodDetail extends mixin(TabsMixin) {
   renderDebugTabView() {
     let {pod} = this.props;
     return (
-      <pre>
-      &lt;PodDebugTabView pod={pod.getId()} /&gt;
-      </pre>
-    );
+        <PodDebugTabView pod={pod} />
```
Comment (mlunoe) [inline src/js/components/PodDetail.js]: #nit indentation

File: src/js/components/PodContainerTerminationTable.js (hunk 2 above)
Comment (mlunoe) [inline src/js/components/PodContainerTerminationTable.js]: done

File: src/js/components/PodDebugTabView.js (hunk 3 above)
Comment (mlunoe) [inline src/js/components/PodDebugTabView.js]: done

File: src/js/components/PodDetail.js (hunk 4 above)
Comment (mlunoe) [inline src/js/components/PodDetail.js]: done

**Comment by mlunoe** [comment]
Let's wait for @leemunroe's approval

**Comment by wavesoft** [comment]
After a discussion with lee yesterday we ended up with something like this:

image

* _No results_: Title `Last Terminations`, text `(No data)`
* _One result_: Title `Last Termination (?? ago)`
* _More than one result_: Append title `Terminated at ??/??/?? (?? ago)` for every other item

**Comment by mlunoe** [comment]
I get this error when I enter the debug tab of a pod that has no tasks:

**Comment by mlunoe** [comment]
Should we handle the error I pointed out, or is this expected?
Could you elaborate a little on the process to come to fix this?
@MatApple Do you know something I don't?

**Comment by MatApple** [comment]
@mlunoe I tested it and didn't get the error.

**Comment by mlunoe** [comment]
I get the error when I use fixtures and enter the debug tab. @wavesoft could you check it out?

**Comment by wavesoft** [comment]
@mlunoe , @MatApple this seems to be unrelated. It looks like the fixtures have not reflected the latest API advancements. If you test it against a live cluster it works fine.

**Comment by mlunoe** [comment]
Right, so we should update the fixtures!

**Pull request merged**